Feature/block7 Set up SonarQube (#25)
devonfw/hangar: documentation/sonarqubedoc/sonarqubeinstance.asciidoc

* `variables.tf` this script contains the variables for the script main.tf.
* `terraform.tfvars` this script contains the parameters for the variables (changeable).

== Usage

Download the files from `...scripts/terrafom/aws/sonarqube`. Open the file `terraform.tfvars` and edit the needed parameters. 

IMPORTANT: If you do not want to change the parameters, you will need a Keypair with the name `sonarqube`.

Run the following script on your command prompt. 

```
execute.sh
```
For inserting new parameters they have to be *Value_name=Parameter* behind execute.sh like in the example below.
```
exaple:  execute.sh var1=par1 var2=par2 ...
```
This script will write the new custom paramaters to the `terraform.tfvars` file. According to this, it will create an ec2 instance with Ubuntu in aws and deploy a Docker container with a sonarqube image on it.  On the command line, you will see the IP address of the ec2 instance.

Once the deployment has finished type `+http://localhost:9000+`(replace localhost with the ec2 instance IP-Address from the output on the command prompt) in your browser and you will have access to your SonarQube. The default values are :

CAUTION: username:   *admin* +

>  password:   *admin*

Reviewer: Still relevant. Terraform command and its explanation should be on the document, not in a separate script. Sorry for not being totally clear.

> == Sonarqube instance in aws 

Reviewer 2: ```suggestion
= Setting up a SonarQube instance in AWS
```

> This repository will show you how to deploy a aws instance with sonarqube on it. This script lauch a sonarqube with an embedded database.

Reviewer 2: ```suggestion
The scope of this section is to deploy an AWS instance with SonarQube. A script has been created in order to assist you in the launching of a SonarQube instance with an embedded database.
```

== Getting Started

> === Prerequisites
> * https://learn.hashicorp.com/collections/terraform/aws-get-started?utm_source=terraform_io&utm_content=terraform_io_footer[terraform]
> 
> * https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-welcome.html[aws cli]

Reviewer 2: ```suggestion
=== Prerequisites
* https://learn.hashicorp.com/collections/terraform/aws-get-started?utm_source=terraform_io&utm_content=terraform_io_footer[Install Terraform].

* https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-welcome.html[Install the AWS CLI].
```

> == available scripts

Reviewer 2: ```suggestion
== The scripts
```

> * `main.tf` the main script wich is deploying the aws Infrastruktur
> * `datascript.sh` install docker on your ec2, an deploy an container wich is runnig sonarqube
> * `variables.tf` here are the Variable for the main.tf
> * `terraform.tfvars` parameters for the variables (changeable)

Reviewer 2: ```suggestion
* `main.tf` this script will deploy the AWS infrastructure.
* `datascript.sh` this script will install Docker on your EC2 instance, and deploy a container running SonarQube.
* `variables.tf` this script contains the variables for the script main.tf.
* `terraform.tfvars` this script contains the parameters for the variables (changeable).
```

> == Using the script

Reviewer 2: ```suggestion
== Usage
```

> Downloade the files from `...scripts/terrafom/aws/sonarqube`. Open the file `terraform.tfvars` and edit the changes of the parameter you need. 

Reviewer 2: ```suggestion
Download the files from `...scripts/terrafom/aws/sonarqube`. Open the file `terraform.tfvars` and edit the needed parameters. 
```

> Downloade the files from `...scripts/terrafom/aws/sonarqube`. Open the file `terraform.tfvars` and edit the changes of the parameter you need. 

Reviewer 2: Add the full path to the scripts.

> IMPORTANT: If you don´t want to chang the Parameters, you will need a Keypair withe the name `sonarqube`.

Reviewer 2: ```suggestion
IMPORTANT: If you do not want to change the parameters, you will need a Keypair with the name `sonarqube`.
```

> Run the following commads on your comand promt.

Reviewer 2: ```suggestion
Run the following commands on your command prompt.
```

```
terraform init
```
This command is used to initialize a working directory containing Terraform configuration files.
```
terraform plan
```
This command creat an execution plan
```
terraform apply --auto-approve
```

> This command run the script and creat ec2 instance with Ubuntu in aws and deploy a docker container with a sonarqube image on it. In the comand promter you will see the Ip-Adress of the ec2 instance.

Reviewer 2: Create a script with this commands so that the user can execute this script and not execute command by command

> After the succesfull deployment type in your Brwoser `+http://localhost:9000+`(replace localhost with the ec2 instance IP-Adress from the output on the comand promt) and you will have acces to sonarqube. The default values are :

Reviewer 2: ```suggestion
Once the deployment has finished type `+http://localhost:9000+`(replace localhost with the ec2 instance IP-Address from the output on the command prompt) in your browser and you will have access to your SonarQube. The default values are :
```

= Setting up a SonarQube instance in AWS
The scope of this section is to deploy an AWS EC2 instance running SonarQube. A script has been created in order to assist you in the launching of a SonarQube instance with an embedded database.

== Getting Started
=== Prerequisites
* https://learn.hashicorp.com/tutorials/terraform/install-cli?in=terraform/aws-get-started[Install Terraform].

* https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html[Install AWS CLI].

== Relevant files

* `main.tf` contains declarative definition written in HCL of AWS infrastructure.
* `setup_sonarqube.sh` script to be run on EC2 instance that installs and deploys a container running SonarQube.
* `variables.tf` contains variable definition for `main.tf`.
* `terraform.tfvars` contains values (user-changeable) for the variables defined in `variables.tf`.
* `addparameter.sh` add values to the terraform.tfvars.

> * `execute.sh` creat the AWS instance with a Docker container wich run SonarQube. 

Reviewer 2: ```suggestion
* `execute.sh` create the AWS EC2 instance with a Docker container which run SonarQube. 
```

== Usage

Download the files from `...scripts/terrafom/aws/sonarqube`. Open the file `terraform.tfvars` and edit the needed parameters. +
For inserting new parameters you can also use `addparameter.sh` followed by a parameter name and value *"--Value_name=Parameter"* like in the example below. +

*EXAMPLE:*
```
addparameter.sh --var1=par1 --var2=par2 ...
```
This script will write the new custom parameters to the `terraform.tfvars` file.  

IMPORTANT: If you do not want to change the parameters, you will need a Keypair with the name `sonarqube`.

Run the following script on your command prompt to deploy SonarQube Instance. 

```
execute.sh
```

> According to this, it will create an ec2 instance with Ubuntu in aws and deploy a Docker container with a SonarQube image on it.  On the command line, you will see the IP address of the EC2 instance.

Reviewer 2: ```suggestion
According to this, it will create an EC2 instance with Ubuntu in AWS and deploy a Docker container with a SonarQube image on it.  On the command line, you will see the IP address of the EC2 instance.
```

= Setting up a SonarQube instance in AWS

The scope of this section is to deploy an AWS EC2 instance running SonarQube. A script has been created in order to assist you in the launching of a SonarQube instance with an embedded database.

== Getting Started
=== Prerequisites
* https://learn.hashicorp.com/tutorials/terraform/install-cli?in=terraform/aws-get-started[Install Terraform].

* https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html[Install AWS CLI].

== Relevant files

* `main.tf` contains declarative definition written in HCL of AWS infrastructure.
* `setup_sonarqube.sh` script to be run on EC2 instance that installs and deploys a container running SonarQube.
* `variables.tf` contains variable definition for `main.tf`.
* `terraform.tfvars` contains values (user-changeable) for the variables defined in `variables.tf`.
* `addparameter.sh` add values to the terraform.tfvars.

> * `execute.sh` create the AWS EC2 instance with a Docker container which run SonarQube. 

Reviewer 2: ```suggestion
```
